src/ImageSharp.Drawing/Processing/PathGradientBrush.cs

```diff
@@ -0,0 +1,219 @@
+                    .First();
+            }
+
+            public Vector4 ColorAt(float distance)
+            {
+                float ratio = this.length > 0 ? distance / this.length : 0;
+
+                return Vector4.Lerp(this.StartColor, this.EndColor, ratio);
+            }
+
+            public Vector4 ColorAt(PointF point) => this.ColorAt(DistanceBetween(point, this.Start));
+        }
+
+        /// <summary>
+        /// The path gradient brush applicator.
+        /// </summary>
+        private class PathGradientBrushApplicator<TPixel> : BrushApplicator<TPixel>
+            where TPixel : struct, IPixel<TPixel>
+        {
+            private readonly Path path;
+
+            private readonly PointF center;
+
+            private readonly Vector4 centerColor;
+
+            private readonly float maxDistance;
+
+            private readonly IList<Edge> edges;
+
+            /// <summary>
+            /// Initializes a new instance of the <see cref="PathGradientBrushApplicator{TPixel}"/> class.
+            /// </summary>
+            /// <param name="source">The source image.</param>
+            /// <param name="path">A polygon that represents the gradient area.</param>
+            /// <param name="edges">Edges of the polygon.</param>
+            /// <param name="centerColor">Color at the center of the gradient area to which the other colors converge.</param>
+            /// <param name="options">The options.</param>
+            public PathGradientBrushApplicator(
+                ImageFrame<TPixel> source,
+                Path path,
+                IList<Edge> edges,
+                Color centerColor,
+                GraphicsOptions options)
+                : base(source, options)
+            {
+                this.path = path;
+                this.edges = edges;
+
+                PointF[] points = path.LineSegments.Select(s => s.EndPoint).ToArray();
+
+                this.center = points.Aggregate((p1, p2) => p1 + p2) / points.Length;
+                this.centerColor = centerColor.ToVector4();
+
+                this.maxDistance = points.Select(p => (Vector2)(p - this.center)).Select(d => d.Length()).Max();
+            }
+
+            /// <inheritdoc />
+            internal override TPixel this[int x, int y]
+            {
+                get
+                {
+                    var point = new PointF(x, y);
+
+                    if (point == this.center)
+                    {
+                        return new Color(this.centerColor).ToPixel<TPixel>();
+                    }
+
+                    if (!this.path.Contains(point))
+                    {
+                        return Color.Transparent.ToPixel<TPixel>();
+                    }
+
+                    Vector2 direction = Vector2.Normalize(point - this.center);
+
+                    PointF end = point + (PointF)(direction * this.maxDistance);
+
+                    (Edge edge, PointF intersection) = this.edges
```

Comment:
Linq here will negatively affect performance.

Reply:
The performance impacts are coming from stuff like extra allocations, virtual calls, deeper stack etc.

Reply:
@antonfirsov Thanks for the clarification. I just want to know exactly how and when using LINQ would affect the performance. I've made this PR to be used for my game project, so the knowledge would benefit me in that aspect too.

I'll try to dig deeper into that LINQ chain to see if there's significant overhead before rewriting it to simple for loops.

Reply:
I'm still not sure if the original code would incur significant overhead, but I just replace it with a plain foreach-loop just in case.

I'm not a seasoned C# developer, so if anyone could help me understand where the performance caveat lies, or how to find out such a problem, it would be much appreciated.